### ISSUE TITLE
Antd Datepicker format value before minDate

### ISSUE THREAD

**Issue by pepe95270** [ant-design/ant-design#50954]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/react-bn65y6?file=demo.tsx)

### Steps to reproduce

Set a DatePicker like this :
```
<DatePicker
    defaultValue={dayjs('2019-09-03', dateFormat)}
    minDate={dayjs('011900', 'MMYYYY')} // <-- You set the minDate to be January 1900
    format={[
      'MMMM YYYY', // <-- You want the date formatted like "January 1995"
      'MMYYYY', // <-- You want the user to be able to write "011995" (but not something before January 1900)
      'DDMMYYYY', // <-- You want the user to be able to write "01011995"
    ]}
  />
```

Then try typping "01011995" (do not copy/paste)
Just after finishing typing "010119" the value is wrongly formated as "January 0119" (so we're unable to finish typing "95")

### What is expected?

It should not transform "010119" to "January 0119" so the user can finish typing a "valid" date like "01011995".

### What is actually happening?

"010119" is wrongly formated to "January 0119" and we're unable to finish writing our date

| Environment | Info |
| --- | --- |
| antd | 5.20.6 |
| React | 18.3.1 |
| System | Windows 10 |
| Browser | Chrome |

---
I know it's a very specific bug but many thanks for the help

**Comment by pepe95270** [comment]
It is kind of related to https://github.com/ant-design/ant-design/issues/48265 but I think it's a different issue.

**Comment by charithmaddoju** [comment]
Hi, I would like to work on this issue. Could you please assign it to me?

**Comment by zombieJ** [comment]
I don't think it should be connect with the logic `minDate` since only part of the date can be limit with `minDate`. e.g.

* **minDate:** 2000/01/01
* **format:** `MMYY` `DDMMYY`
* **type:** 010100

Both typing is valid with the `format`: 
* 0101: 2001/01
* 010100: 2000/01/01

It should be to provide a `parser` api instead of connect with `minDate` logic.

**Comment by zombieJ** [comment]
@charithmaddoju, pls feel free to PR directly.

**Comment by pepe95270** [comment]
A parser API is a very good idea yes 👍 

But I think it could also be good to add this behaviour to the default parser.
Why format an invalid date (for example a date before minDate) ?
